selfdrive/car/subaru/carstate.py

```diff
@@ -152,5 +160,13 @@ def update(self, cp, cp_cam):
       cp.vl["BodyInfo"]['DOOR_OPEN_FR'],
       cp.vl["BodyInfo"]['DOOR_OPEN_FL']])
 
-    self.es_distance_msg = copy.copy(cp_cam.vl["ES_Distance"])
-    self.es_lkas_msg = copy.copy(cp_cam.vl["ES_LKAS_State"])
+    if self.car_fingerprint == CAR.IMPREZA:
+      self.v_cruise_pcm = cp_cam.vl["ES_DashStatus"]["Cruise_Set_Speed"] * CV.MPH_TO_KPH
+      self.es_distance_msg = copy.copy(cp_cam.vl["ES_Distance"])
+      self.es_lkas_msg = copy.copy(cp_cam.vl["ES_LKAS_State"])
+      # 1 = imperial, 6 = metric
+      if cp.vl["Dash_State"]['Units'] == 1:
+        self.v_cruise_pcm *= CV.MPH_TO_KPH     
+    else:
+      self.v_cruise_pcm = cp_cam.vl["ES_DashStatus"]["Cruise_Set_Speed"]
+      self.steer_not_allowed = cp.vl["Steering_Torque"]["LKA_Lockout"]
```

Comment:
this breaks the Impreza. All the cars shall have the same attributes.

Reply:
Which lines break the Impreza, 163-169? Should be the same. I'll look into it more later. The Outback and Legacy are on a very different platform

Reply:
I'll change that else to outback and legacy

Reply:
steer_not_allowed is used in interface. so needs to be defined for all cars

Reply:
Ah I see now, I'll fix these all tonight. Thanks

Reply:
I made the changes
Would you like me to redo this on 0.6?